File: Data Structures & Algorithms/two-integer-sum/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        const mp = new Map<number, number>();

    for(let i = 0; i < nums.length; i++){
        const need = target - nums[i];

        if(mp.has(need)){
            return [mp.get(need)!, i];
        }

        mp.set(nums[i], i);
    }

    return [];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
       int  a=0,b=numbers.size()-1;
       while(numbers[a]+numbers[b]!=target){
        if(numbers[a]+numbers[b]<target){
            a+=1;
        } else if(numbers[a]+numbers[b]>target){
            b-=1;
        }
       }
       return {a+1,b+1};
    }
};
